src/measurement_kit/main.cpp

```diff
@@ -153,10 +153,12 @@ int main(int argc, char **argv) {
     /*
      * Allow to call getopt() again.
      *
-     * Non portable. Assume it's either GNU or BSD. We can do better in
-     * configure checking for the proper way to reset options.
+     * Non portable. Verify using configure that we have `optreset` (which
+     * we also have on Windows where we embed a getopt() implementation
+     * that features such flag). Otherwise check whether we're in the GNU
+     * C library, which is a different behaviour. Otherwise bail.
      */
-#if HAVE_DECL_OPTRESET
+#if HAVE_DECL_OPTRESET || defined _WIN32
```

Comment:
TODO(@bassosimone): IIRC CMakeLists.txt has a check for optreset, so we can probably avoid the `|| defined _WIN32`.